misc/mzutil/await_view_states.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+"""await_view_states
+
+Script to query all materialized views in a Materialize database and wait until each view matches
+the exact output as captured in the snapshot files.
+
+Prints timing information to indicate how long each view took to reach the desired state.
+"""
+
+import argparse
+import glob
+import io
+import os
+import time
+
+import psycopg2
+import psycopg2.errors
+
+
+def view_names(conn):
+    """Return a generator containing all view names in Materialize."""
+    with conn.cursor() as cursor:
+        cursor.execute("SHOW VIEWS")
+        for row in cursor:
+            yield row[0]
+
+
+def view_matches(cursor, view, expected):
+    """Return True if a SELECT from the VIEW matches the expected string."""
+    stream = io.StringIO()
+    try:
+        cursor.copy_expert(f"COPY (SELECT * FROM {view}) TO STDOUT", stream)
+    except psycopg2.errors.InternalError_:
+        # The view is not yet ready to be queried
+        return False
+    return stream.getvalue() == expected
+
+
+def await_materialize_views(args):
+    """Record the current table status of all views installed in Materialize."""
+
+    start_time = time.time()
+
+    def file_contents(fname):
+        with open(fname, "r") as fd:
+            return fd.read()
+
+    # Create a dictionary mapping view names (as calculated from the filename) to expected contents
+    view_snapshots = {
+        os.path.splitext(os.path.basename(fname))[0]: file_contents(fname)
+        for fname in glob.glob(os.path.join(args.snapshot_dir, "*.sql"))
+    }
```

Comment:
So this is fine, but I think it'd be easier with [pathlib](https://docs.python.org/3/library/pathlib.html):

```suggestion
from pathlib import Path

snapshots = {p.stem: p.read_text() for p in Path(args.snapshot_dir).glob("*.sql")}
```